filebeat/input/filestream/internal/input-logfile/manager.go

```diff
@@ -167,18 +167,21 @@ func (cim *InputManager) Create(config *conf.C) (v2.Input, error) {
 		return nil, err
 	}
 
+	defer func() {
+		//If there is any error creating the input, remove it from the IDs list
+		if retErr != nil {
+			cim.idsMux.Lock()
+			delete(cim.ids, settings.ID)
```

Comment:
What happens if `settings.AllowIDDuplication` is true, and the first id is fine, but the second throws an error.  Is it still correct to delete the ID?

I think the issue is that there are multiple Lock/UnLock calls and you really don't know if the data associated with `settings.ID` truly maps to the retErr.  Another go routine could have updated the data with something valid in between.